zerver/views/auth.py

```diff
@@ -933,6 +919,96 @@ def api_fetch_api_key(
     RequestNotes.get_notes(request).requestor_for_logs = user_profile.format_requestor_for_logs()
 
     api_key = get_api_key(user_profile)
+    return api_key
+
+
+@csrf_exempt
+@has_request_variables
+def jwt_fetch_api_key(
+    request: HttpRequest,
+    json_web_token: str = REQ(default=""),
+    include_profile: bool = REQ(default=False, json_validator=check_bool),
+) -> HttpResponse:
+    return_data: Dict[str, bool] = {}
+
+    realm = get_realm_from_request(request)
+    if realm is None:
+        raise InvalidSubdomainError()
+
+    try:
+        key = settings.JWT_AUTH_KEYS[realm.subdomain]["key"]
+        algorithms = settings.JWT_AUTH_KEYS[realm.subdomain]["algorithms"]
+    except KeyError:
+        raise JsonableError(_("Auth key for this subdomain not found"))
+
+    try:
+        json_web_token = request.POST["json_web_token"]
```

Comment:
We should be using the parameter from REQ here.

Also, is `json_web_token` a reasonably standard name for a JWT token parameter? I feel like I'd expect something more like `jwt` or `token` or something, but have no experience with JWT personally.

Reply:
Hmm yeah a generic `token` seems more standard, tweaked that + added REQ + `@require_post`